#!/usr/bin/env python
PACKAGE = "serl_franka_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness", 2000, 0, 6000)
gen.add("translational_damping", double_t, 0, "Cartesian translational damping", 89, 0, 400)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness", 150, 0, 300)
gen.add("rotational_damping", double_t, 0, "Cartesian rotational damping", 7, 0, 30)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.2, 0, 100)
gen.add("joint1_nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 100, 0, 100)
gen.add("translational_clip_neg_x", double_t, 0, "Value to clip error to in realtime control loop", 0.01, 0, 0.1)
gen.add("translational_clip_neg_y", double_t, 0, "Value to clip error to in realtime control loop", 0.01, 0, 0.1)
gen.add("translational_clip_neg_z", double_t, 0, "Value to clip error to in realtime control loop", 0.01, 0, 0.1)
gen.add("translational_clip_x", double_t, 0, "Value to clip error to in realtime control loop", 0.01, 0, 0.1)
gen.add("translational_clip_y", double_t, 0, "Value to clip error to in realtime control loop", 0.01, 0, 0.1)
gen.add("translational_clip_z", double_t, 0, "Value to clip error to in realtime control loop", 0.01, 0, 0.1)
gen.add("rotational_clip_neg_x", double_t, 0, "Value to clip error to in realtime control loop", 0.05, 0, 0.1)
gen.add("rotational_clip_neg_y", double_t, 0, "Value to clip error to in realtime control loop", 0.05, 0, 0.1)
gen.add("rotational_clip_neg_z", double_t, 0, "Value to clip error to in realtime control loop", 0.05, 0, 0.1)
gen.add("rotational_clip_x", double_t, 0, "Value to clip error to in realtime control loop", 0.05, 0, 0.1)
gen.add("rotational_clip_y", double_t, 0, "Value to clip error to in realtime control loop", 0.05, 0, 0.1)
gen.add("rotational_clip_z", double_t, 0, "Value to clip error to in realtime control loop", 0.05, 0, 0.1)
gen.add("translational_Ki", double_t, 0, "Cartesian translational intergral gain", 0, 0, 100)
gen.add("rotational_Ki", double_t, 0, "Cartesian rotational intergral gain", 0, 0, 100)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
